File: PortfolioManager/PortfolioManager.Common/DTO/StockDTO.cs
using System.ComponentModel.DataAnnotations;

namespace PortfolioManager.Common.DTO;

public class StockDTO
{
    public int Id { get; set; }
    [MaxLength(10)]
    public string Symbol { get; set; } = string.Empty;
    [MaxLength(200)]
    public string Name { get; set; } = string.Empty;
    public int Quantity { get; set; }
    public double EntryPrice { get; set; }
    public double CurrentPrice { get; set; }    
    public double TotalCurrentPrice { get; set; } //Quantity * CurrentPrice
    public double Gain { get; set; } //Quantity * CurrentPrice - Quantity * EntryPrice
    public double GainPercentage { get; set; } //Gain / (EntryPrice * Quantity) * 100
    public DateTime EntryDate { get; set; }
    [MaxLength(10)]
    public string Currency { get; set; } = string.Empty; 
    public SectorEnum Sector { get; set; } 
    public int PortfolioId { get; set; }
    
}

File: PortfolioManager/PortfolioManager.Client/portfolio-manager-client/src/DTO/StockDto.ts
export default interface StockDto {
    id: number;
    symbol: string;
    name: string;
    quantity: number;
    entryPrice: number;
    currentPrice: number;
    totalCurrentPrice: number;
    gain: number;
    gainPercentage: number;
    entryDate: Date;
    currency: string;
    sector: SectorEnum;
    portfolioId: number;
}

export const getSectorText = (sector: number): string => {
    switch (sector) {
        case SectorEnum.Technology:
            return "Technology";
        case SectorEnum.FinancialServices:
            return "Financial Services";
        case SectorEnum.Healthcare:
            return "Healthcare";
        case SectorEnum.ConsumerCyclical:
            return "Consumer Cyclical";
        case SectorEnum.Industrial:
            return "Industrial";
        case SectorEnum.CommunicationServices:
            return "Communication Services";
        case SectorEnum.ConsumerDefensive:
            return "Consumer Defensive";
        case SectorEnum.Energy:
            return "Energy";
        case SectorEnum.BasicMaterials:
            return "Basic Materials";
        case SectorEnum.RealEstate:
            return "Real Estate";
        case SectorEnum.Utilities:
            return "Utilities";
        default:
            return "Unknown";
    }
}

export enum SectorEnum {
    Technology = 0,
    FinancialServices = 1,
    Healthcare = 2,
    ConsumerCyclical = 3,
    Industrial = 4,
    CommunicationServices = 5,
    ConsumerDefensive = 6,
    Energy = 7,
    BasicMaterials = 8,
    RealEstate = 9,
    Utilities = 10
}
